setuptools/command/egg_info.py

```diff
@@ -341,38 +567,13 @@ def add_defaults(self):
         elif os.path.exists(self.manifest):
             self.read_manifest()
         ei_cmd = self.get_finalized_command('egg_info')
-        self._add_egg_info(cmd=ei_cmd)
-        self.filelist.include_pattern("*", prefix=ei_cmd.egg_info)
-
-    def _add_egg_info(self, cmd):
```

Comment:
The behaviour is still required, but is now as simple as `self.filelist.graft(ei_cmd.egg_info)`, assuming I understand what is going in `_add_egg_info()`. I've replaced that whole function with that one-liner.